examples/bigquery_user_info_updater_tool/cron/bigquery_user_info_updater/tests/test_nested_user_info_updater.py

```diff
@@ -0,0 +1,255 @@
+        temp_table_query = self.bq_client.query(
+            query='SELECT * FROM `{0:s}.{1:s}.{2:s}`'.format(
+                project_id,
+                self.dataset_id,
+                self.temp_user_info_updates_id
+            ),
+            job_config=temp_table_query_config,
+            location='US'
+        )
+        temp_table_query.result()
+        temp_table_results_df = temp_table_query.to_dataframe() \
+            .sort_values(by=['userId']).reset_index(drop=True)
+
+        # Gather expected results for comparison
+        abs_path = os.path.abspath(os.path.dirname(__file__))
+        expected_temp_data_file = os.path.join(
+            abs_path,
+            'test_data/nested_data/temp_user_info_updates_expected.json'
+        )
+        expected_temp_table_df = pd.read_json(
+            expected_temp_data_file
+        )
+        # Reorder columns since read_json() reads them alphabetically
+        with open(self.schema_path, 'r') as f:
+            json_schema = json.loads(f.read())
+        col_list = [str(col['name']) for col in json_schema['fields']]
+        expected_temp_table_df = expected_temp_table_df[col_list]
+
+        # convert ingestTimestamp to datetime
+        expected_temp_table_df['ingestTimestamp'] = pd.to_datetime(
+            expected_temp_table_df['ingestTimestamp']
+        )
+
+        # Compare results
+        pd.testing.assert_frame_equal(
+            temp_table_results_df,
+            expected_temp_table_df
+        )
+
+        # Query the final table to test that the merge_updates() function worked
+        final_table_query_config = bigquery.QueryJobConfig()
+        final_table_query_config.use_legacy_sql = False
+        final_table_query = self.bq_client.query(
+            query='SELECT * FROM `{0:s}.{1:s}.{2:s}`'.format(
+                project_id,
+                self.dataset_id,
+                self.user_info_final_id
+            ),
+            job_config=final_table_query_config,
+            location='US'
+        )
+        final_table_query.result()
+        final_table_results_df = final_table_query.to_dataframe() \
+            .sort_values(by=['userId']).reset_index(drop=True)
+        # Gather expected results for comparison
+        expected_final_data_file = os.path.join(
+            abs_path,
+            'test_data/nested_data/user_info_final_expected.json'
+        )
+        expected_final_table_df = pd.read_json(expected_final_data_file)
+
+        # Reorder columns since read_json() reads them alphabetically
+        with open(self.schema_path, 'r') as f:
+            json_schema = json.loads(f.read())
+        col_list = [str(col['name']) for col in json_schema['fields']]
+        expected_final_table_df = expected_final_table_df[col_list]
+
+        # convert ingestTimestamp to datetime
+        expected_final_table_df['ingestTimestamp'] = pd.to_datetime(
+            expected_final_table_df['ingestTimestamp']
+        )
+
+        # Compare results
```

Comment:
jaketf on Aug 27, 2019 Member
see note on other test about pandas utility for this.